source/slang/slang-check-constraint.cpp

```diff
@@ -158,6 +158,11 @@ namespace Slang
                 return bestType;
         }
 
+        if (auto extractExistentialType = as<ExtractExistentialType>(type))
+        {
+            if (extractExistentialType->interfaceDeclRef.equals(interfaceDeclRef))
+                return extractExistentialType;
+        }
```

Comment:
Hmm... I would have expected this to be a check that `type` is a `DeclRefType` referring to an `InterfaceDecl` and with a `ThisTypeSubstitution`, and then it would `return` the underlying type from the `ThisTypeSubstitution`.